src/sagemaker/workflow/steps.py

```diff
@@ -358,16 +360,25 @@ def arguments(self) -> RequestType:
         ModelName cannot be included in the arguments.
         """
 
-        request_dict = self.model.sagemaker_session._create_model_request(
-            name="",
-            role=self.model.role,
-            container_defs=self.model.prepare_container_def(
-                instance_type=self.inputs.instance_type,
-                accelerator_type=self.inputs.accelerator_type,
-            ),
-            vpc_config=self.model.vpc_config,
-            enable_network_isolation=self.model.enable_network_isolation(),
-        )
+        if isinstance(self.model, PipelineModel):
+            request_dict = self.model.sagemaker_session._create_model_request(
+                name="",
+                role=self.model.role,
+                container_defs=self.model.pipeline_container_def(self.inputs.instance_type),
+                vpc_config=self.model.vpc_config,
+                enable_network_isolation=self.model.enable_network_isolation,
+            )
+        else:
+            request_dict = self.model.sagemaker_session._create_model_request(
+                name="",
+                role=self.model.role,
+                container_defs=self.model.prepare_container_def(
+                    instance_type=self.inputs.instance_type,
+                    accelerator_type=self.inputs.accelerator_type,
```

Comment:
nit: we could save some redundant code by only assigning `accelerator_type` over the if else block.

Reply:
+1, can we collapse this into a single call but do the conditional check only in the accelerator_type param

```
accelerator_type=self.inputs.accelerator_type if isinstance(self.model, PipelineModel) else None